### PR TITLE
Code coverage: Merge expenses service - 20%

### PR DESCRIPTION
**Pull request by Dimple16**
Screenshot 2023-03-06 at 11 58 45 PM

### CODE REVIEW

File: src/app/core/models/expense.model.ts [hunk 1]
```diff
@@ -207,4 +207,5 @@ export interface Expense {
   debit?: boolean;
   duplicates?: { fields: string[]; percent: number; reason: string; transaction_id: string }[];
   tx_is_split_expense?: boolean;
+  custom_fields?: { [key: string]: any };
```
Comment (rvab) [inline src/app/core/models/expense.model.ts:210]: Don't we have an interface for `custom_fields`?

File: src/app/core/services/merge-expenses.service.spec.ts [hunk 2]
```diff
@@ -93,4 +112,167 @@ describe('MergeExpensesService', () => {
...
+    it('should return empty list if advance expense is not present', () => {
+      expect(mergeExpensesService.checkIfAdvanceExpensePresent(apiExpenseRes)).toEqual([]);
+    });
+
+    it('should return empty list if expense list is empty', () => {
+      expect(mergeExpensesService.checkIfAdvanceExpensePresent([null])).toEqual([]);
+    });
+  });
+
+  it('mergeExpenses(): should merge the expenses', () => {
```
Comment (rvab) [inline src/app/core/services/merge-expenses.service.spec.ts:144]: `done` is missing

File: src/app/core/services/merge-expenses.service.spec.ts [hunk 3]
```diff
@@ -93,4 +112,167 @@ describe('MergeExpensesService', () => {
...
+      expect(mergeExpensesService.formatPaymentModeOptions({ value: AccountType.CCC })).toEqual(paymentModeOptions3);
+    });
+  });
+
+  it('checkOptionsAreSame(): should return true if options are same', () => {
+    // @ts-ignore
+    expect(mergeExpensesService.checkOptionsAreSame(sameOptions)).toBeTrue();
+  });
+
+  it('formatOptions(): should return the formatted options', () => {
```
Comment (rvab) [inline src/app/core/services/merge-expenses.service.spec.ts:252]: `done` is missing

**Review by rvab — APPROVED** [review]
Minor comments

File: src/app/core/services/merge-expenses.service.spec.ts (hunk 3 above)
Comment (Dimple16) [reply src/app/core/services/merge-expenses.service.spec.ts:252]: this doesn't deal with observables

File: src/app/core/models/expense.model.ts (hunk 1 above)
Comment (Dimple16) [reply src/app/core/models/expense.model.ts:210]: that is `custom_properties`
this custom field is a key that has the custom field display name mapped to its value

**Pull request merged**